src/gwt/src/org/rstudio/studio/client/workbench/views/output/find/FindOutputPresenter.java

```diff
@@ -143,13 +191,192 @@ public void onFindOperationEnded(
          {
             if (event.getHandle() == currentFindHandle_)
             {
-               currentFindHandle_ = null;
+               if (!view_.getReplaceMode())
+                  currentFindHandle_ = null;
                view_.setStopSearchButtonVisible(false);
                view_.showSearchCompleted();
             }
          }
       });
 
+      events_.addHandler(PreviewReplaceEvent.TYPE, new PreviewReplaceEvent.Handler()
+      {
+         @Override
+         public void onPreviewReplace(PreviewReplaceEvent event)
+         {
+            view_.setRegexPreviewMode(true);
+            stopAndClear();
+            dialogState_.clearResultsCount();
+
+            FileSystemItem searchPath =
+                                      FileSystemItem.createDir(dialogState_.getPath());
+            JsArrayString filePatterns = JsArrayString.createArray().cast();
+            for (String pattern : dialogState_.getFilePatterns())
+               filePatterns.push(pattern);
+
+            server_.previewReplace(dialogState_.getQuery(),
+                                   dialogState_.isRegex(),
+                                   !dialogState_.isCaseSensitive(),
+                                   searchPath,
+                                   filePatterns,
+                                   view_.getReplaceText(),
+                                   view_.isReplaceRegex(),
+                                   view_.useGitIgnore(),
+                                   new SimpleRequestCallback<String>()
+                                   {
+                                      @Override
+                                      public void onResponseReceived(String handle)
+                                      {
+                                         view_.clearMatches();
+                                         currentFindHandle_ = handle;
+                                         dialogState_.clearResultsCount();
+                                      }
+                                   });
+         }
+      });
+
+      view_.getStopReplaceButton().addClickHandler(new ClickHandler()
+      {
+         @Override
+         public void onClick(ClickEvent event)
+         {
+            globalDisplay_.showYesNoMessage(
+                  GlobalDisplay.MSG_WARNING,
+                  "Stop Replace",
+                  "Are you sure you want to cancel the replace? Changes already made will not be reverted.",
```

Comment:
Currently the operation continues, but I think I want to change this since in most cases the replace operation will be very quick and a few seconds/milliseconds makes a difference.

Reply:
Created project card for this.